### PR TITLE
[16.0][IMP] fastapi: add company_id dependency override

### PR DESCRIPTION
**Pull request by marielejeune**
The company can be specified on the `fastapi.endpoint`

### CODE REVIEW

**Comment by marielejeune** [comment]
@lmignon @sbidoul @AnizR Here is a new dependency override to set the company on the fastapi endpoint. 

I added this to https://github.com/shopinvader/odoo-shopinvader/pull/1364 to be able to play with it.

**Review by lmignon — APPROVED** [review]
Thank you @marielejeune LGTM.

File: fastapi/dependencies.py [hunk 1]
```diff
@@ -19,8 +19,19 @@
     from .models.fastapi_endpoint import FastapiEndpoint
 
 
-def odoo_env() -> Environment:
-    yield odoo_env_ctx.get()
+def company_id() -> int | None:
+    """This method is overriden by the FastAPI app to make the company
+    available for your endpoint.
```
Comment (sbidoul) [inline fastapi/dependencies.py]:
```suggestion
    """This method may be overriden by the FastAPI app to define the allowed company
    in the Odoo environment of the endpoint. By default, the company defined on the endpoint record
    is used.
```

File: fastapi/dependencies.py [hunk 2]
```diff
@@ -19,8 +19,20 @@
...
+    """This method may be overriden by the FastAPI app to make the company
+    available for your endpoint. By default, the company defined on the
+    endpoint record is used.
+    """
+    return None
+
+
+def odoo_env(company_id: Annotated[int | None, Depends(company_id)]) -> Environment:
+    env = odoo_env_ctx.get()
+    if company_id:
```
Comment (sbidoul) [inline fastapi/dependencies.py]:
```suggestion
    if company_id is not None:
```

Pas grave dans ce cas mais une bonne habitude à prendre (pour les situations où on veut distinguer 0 et None).

File: fastapi/dependencies.py [hunk 3]
```diff
@@ -19,8 +19,20 @@
     from .models.fastapi_endpoint import FastapiEndpoint
 
 
-def odoo_env() -> Environment:
-    yield odoo_env_ctx.get()
+def company_id() -> int | None:
+    """This method may be overriden by the FastAPI app to make the company
+    available for your endpoint. By default, the company defined on the
+    endpoint record is used.
```
Comment (sbidoul) [inline fastapi/dependencies.py]:
```suggestion
    """This method may be overriden by the FastAPI app to set the allowed company
    in the Odoo env of the endpoint. By default, the company defined on the
    endpoint record is used.
```

**Review by AnizR — APPROVED** [review]
Great,Thanks :smile:

File: fastapi/models/fastapi_endpoint.py [hunk 4]
```diff
@@ -85,6 +92,11 @@ def _compute_urls(self):
             rec.redoc_url = f"{rec.root_path}/redoc"
             rec.openapi_url = f"{rec.root_path}/openapi.json"
 
+    @api.depends("user_id")
+    def _compute_company_id(self):
+        for endpoint in self:
+            endpoint.company_id = endpoint.user_id.company_id
```
Comment (sbidoul) [inline fastapi/models/fastapi_endpoint.py:98]: A little style remark: I think `for rec in self` or `for record in self`, when done consistently is easier to read than `for endpoint in self`. Just like self is... self, `rec` or `record`  for the current record in the self iterator.

**Pull request merged**